[WIP] Enhancement of sharing with OpenTheory

francoisthire opened the pull request:
I open this PR just to follow the diff with the current version of Holide.

Commit:
work in progress

francoisthire commented on src/article.ml:
```diff
@@ -84,14 +92,17 @@ let process_command cmd stack =
         obj :: stack
-      | "defineConst", Term(t) :: Name(n) :: stack -> Thm(Thm.define_const n t) :: Const(n) :: stack
+      | "defineConst", Term(t) :: Name(n) :: stack ->
+		Term.add_cst n (Term.type_of t);
```
Why did you add this? @EmilieGrienenberger

EmilieGrienenberger replied:
The constants are added to a table, the goals being to remember where they are defined and ultimately to be able to share them between the generated dk files.

francoisthire commented on src/input.ml:
```diff
@@ -11,7 +11,7 @@ let set_input filename =
 (** Get the name of the module from the name of the file. *)
 let get_module_name () =
-  Filename.chop_extension (Filename.basename !input_file)
+  Output.low_dash (Filename.chop_extension (Filename.basename !input_file))
```
What is this function "low_dash"?

EmilieGrienenberger replied:
The goal is to avoid "Invalid name directive ignored" warning : usual OpenTheory filenames have "-" characters, which were transformed while writing to Dedukti.

Instead of changing the filenames similarly, I chose to transform "-" to low dashes to maintain readability.

Commit:
Adding close_in

Commit:
Adding dependencies to database, topological sort

Commit:
Adding debugging printing for types

Commit:
For ocamlgraph

Commit:
Closing input files when reaching end of file

Commit:
- Fixing bug on free type variables when using EqMp rule
- Making named theorems more readable

Commit:
Removing #NAME command and adding #REQUIRE commands.

Commit:
- Making #REQUIRES more readable (less blank lines)
- Correcting bug on auto-dependent modules
	(due to lack of definition of type operators' constants)

Commit:
- Adding local defs
- Deactivate sharing in statement of main definitions/theorems/declarations

Commit:
Removing printing of local keyword

Commit:
Differentiate sorting and database, start work to construct iteratively constructed database

Commit:
Axiomatized connectives

Commit:
Rename Database module to Sort, use axiomatized connectives, locally deactivate sharing when possible

Commit:
Dk files for axiomatized connectives

Commit:
Ignoring compilation byproducts

Commit:
using dkcheck --version instead of dkcheck -version

Commit:
gilbert connective

Commit:
Change build directory

Commit:
Structure of repo for merlin

Commit:
Change build directory

Commit:
Remove debugging, less verbose printing

Commit:
Work on gilbert connectives

Commit:
work on holide object database

Commit:
improved version fo gilbert connective
